app/javascript/packs/articlePage.jsx

```diff
@@ -90,6 +90,9 @@ getCsrfToken().then(async () => {
   const root = document.getElementById('comment-subscription');
   const isLoggedIn = userStatus === 'logged-in';
 
+  if (!root) {
```

Comment:
There are 7 published articles in dev where show comments is false - and we only include the comment-subscription div if we're showing comments.

It's possible this is not the only reason that the div would be missing, but we're seeing failures in the error handler since we try to write to root's innerHTML in the catch block.

Reply:
I think the `articles.show_comments` column is sort of a legacy column, seeing as those 7 articles were published around the inception of dev.to. Maybe we should update those articles to `show_comments = true` instead?

Reply:
actually - if we're going to do that - it probable makes _more_ sense to remove the conditional guards around this behavior from the view (it's part of both the cache key, and the conditional content of the "comments" section)

https://github.com/forem/forem/blob/main/app/views/articles/_full_comment_area.html.erb#L1-L3

Reply:
Ahh good point. Hmm will have to ping @benhalpern on this decision.

Reply:
> it probable makes more sense to remove the conditional guards around this behavior from the view

That's logically true, but I think then we will see the same issues with the Cypress test this cropped up in. Unlike 'normal' users, Cypress is sometimes moving off of this page _super_ quickly, after this pack has been triggered, but before this code has fully run. So basically, it's assigning `const root` after the test has already moved away from the page, making the `root` `null`.

So I think in any case we probably need to keep this defensive coding 😕

Reply:
ah - my comment (about removing the guard clause) was referring to the `if @article.show_comments` conditional rendering in the view, not about the "check if we found a comment-subscription div when we called getElementById" added in this PR.

If we're removing the ability to have articles with comments disabled (which only applies to 7 published articles in DEV from 2016-2017), then there's no need to maintain conditional rendering in the ruby view template for comments, always include the comment-subscription div.

However, it probably _also_ makes sense to leave the guard around root being null here, in case the page changed/cleared very quickly.

Reply:
Ah I get you! Yeah that all sounds sensible 😄